### PR TITLE
[XGBoost] add XGBoost inference support

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Add support for XGBoost model. Now only supports for Mac and Linux.

### CODE REVIEW

File: ml/xgboost/src/main/java/ai/djl/ml/xgboost/XgbNDArray.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void close() {
+        if (handle != null && handle.get() != 0L) {
+            long pointer = handle.getAndSet(0L);
+            JniUtils.deleteDMatrix(pointer);
+        }
+        if (data != null) {
```
Comment (frankfliu) [inline ml/xgboost/src/main/java/ai/djl/ml/xgboost/XgbNDArray.java:154]: You don't need to check null

File: ml/xgboost/src/test/java/ai/djl/ml/xgboost/XgbModelTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class XgbModelTest {
+
+    @BeforeClass
+    public void downloadXGBoostModel() throws IOException {
+        Path modelDir = Paths.get("build/model");
+        DownloadUtils.download(
+                "https://resources.djl.ai/test-models/xgboost/regression.json",
```
Comment (frankfliu) [inline ml/xgboost/src/test/java/ai/djl/ml/xgboost/XgbModelTest.java:39]: how big is the json file? better check in to test/resources folder.

**Review by frankfliu — APPROVED** [review]

File: ml/xgboost/src/test/java/ai/djl/ml/xgboost/XgbModelTest.java (hunk 2 above)
Comment (lanking520) [reply ml/xgboost/src/test/java/ai/djl/ml/xgboost/XgbModelTest.java:39]: Around 114kb

File: ml/xgboost/src/main/java/ai/djl/ml/xgboost/XgbNDArray.java (hunk 1 above)
Comment (lanking520) [reply ml/xgboost/src/main/java/ai/djl/ml/xgboost/XgbNDArray.java:154]: Then we essentially not collecting garbage here?

**Pull request merged**